cmd/yggdrasil/chuser_unix.go

```diff
@@ -0,0 +1,87 @@
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris
+// +build aix darwin dragonfly freebsd linux netbsd openbsd solaris
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	osuser "os/user"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+func chuser(user string) error {
+	group := ""
+	if i := strings.IndexByte(user, ':'); i >= 0 {
+		user, group = user[:i], user[i+1:]
+	}
```

Comment:
Why not just `strings.Split(user, ':')`?

Reply:
`IndexByte` is probably speediest as `Split` allocates slice and uses string for splitting.
`Cut` would probably be more fitting here API niceness wise, but I'm not sure if really worth changing at this point.

Reply:
> `IndexByte` is probably speediest as `Split` allocates slice and uses string for splitting.

This code runs once, it is not a hot path.

> `Cut` would probably be more fitting here API niceness wise, but I'm not sure if really worth changing at this point.

Sure, either way looks cleaner than manual indexing.